ios/BurntModule.swift

```diff
@@ -107,118 +127,146 @@ struct ToastOptions: Record {
     }
   }
 }
 
-enum ToastPresentSide: String, EnumArgument {
+enum ToastPresentSide: String, Enumerable {
   case top
   case bottom
-
+  
   func toSPIndicatorPresentSide() -> SPIndicatorPresentSide {
     switch self {
-    case .top:
-      return .top
-    case .bottom:
-      return .bottom
+      case .top:
+        return .top
+      case .bottom:
+        return .bottom
     }
   }
 }
 
 public class BurntModule: Module {
   public func definition() -> ModuleDefinition {
     Name("Burnt")
-
+    
     AsyncFunction("toastAsync") { (options: ToastOptions) -> Void in
-      let view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset())
-
+      let view : SPIndicatorView
+      if(options.preset == .none){
+        view = SPIndicatorView(title: options.title, message: options.message)
+      } else if(options.preset == .custom){
+        view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset(options)!)
+      }
+      else{
+        view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset(nil)!)
+      }
+      
       if let duration = options.duration {
         view.duration = duration
       }
-
+      
       if let icon = options.layout?.iconSize {
         view.layout.iconSize = .init(width: icon.width, height: icon.height)
       }
-
+      
       view.dismissByDrag = options.shouldDismissByDrag
-
+      
       view.presentSide = options.from.toSPIndicatorPresentSide();
-
+      
       view.present(haptic: options.haptic.toSPIndicatorHaptic())
-    }.runOnQueue(.main) 
-
+    }.runOnQueue(.main)
+    
     AsyncFunction("alertAsync")  { (options: AlertOptions) -> Void in
-      let view = SPAlertView(
-        title: options.title, 
-        message: options.message, 
-        preset: options.preset.toSPAlertIconPreset())
-
-        if let duration = options.duration {
-          view.duration = duration
-        }
-
-        view.dismissByTap = options.shouldDismissByTap
-
-        if let icon = options.layout?.iconSize {
-          view.layout.iconSize = .init(width: icon.width, height: icon.height)
-        }
-
-        view.present(
-          haptic: options.haptic.toSPAlertHaptic())
-     }.runOnQueue(.main) 
-
+      let view : SPAlertView
+      if(options.preset == .custom){
+        view = SPAlertView(
+          title: options.title,
+          message: options.message,
+          preset: options.preset.toSPAlertIconPreset(options))
+      }
+      else{
+        view = SPAlertView(
+          title: options.title,
+          message: options.message,
+          preset: options.preset.toSPAlertIconPreset(nil))
+      }
+      
+      
+      if let duration = options.duration {
+        view.duration = duration
+      }
+      
+      view.dismissByTap = options.shouldDismissByTap
+      
+      if let icon = options.layout?.iconSize {
```

Comment:
Should this get removed now? But btw, this is what I meant with the unwrapping.

Reply:
no, it can be used to resize ICONS and I think it's useful.

Reply:
But I thought this got moved to `ios.icon.width` and `ios.icon.height`?

Reply:
It's still reading `options.layout.iconSize`, that's what I mean. It should be `if let iconWidth = options.icon?.width`... right?

Reply:
See my comment [here](https://github.com/nandorojo/burnt/pull/12/files/2d6e8e50c6fd63b97bbe7a585c4ebc6ebb95385b#r1124960565).

Reply:
> But I thought this got moved to ios.icon.width and ios.icon.height? 

🤣 sorry, I didn't make that change, just kept the before way because the `ios.icon` object is only available in the "custom" preset.

Reply:
Let's remove these [fields](https://github.com/nandorojo/burnt/pull/12/files#diff-3ee7f7cceac93ff6a6567332ccd4b93afc832580787dec814b8a538234452796R85-R88) then.

Reply:
oh yeah, that's my bad forget to remove it.

Reply:
I just removed them!